### PR TITLE
Added CoreCLR APIs for Nancy.Testing

### PR DESCRIPTION
**Pull request by jchannon**
This makes Nancy.Testing dotnet5.4 compatible apart from some areas:

- [x] Nancy.Forms.Authentication
- [x] CSQuery
- [x] `Assembly.GetEntryAssembly`

### CODE REVIEW

File: src/Nancy.Testing/BrowserContext.cs [hunk 1]
```diff
@@ -202,19 +203,24 @@ public void HostName(string hostName)
...
+                            ZS1TdGF0ZTEOMAwGA1UEChMFTmFuY3kxDjAMBgNVBAsTBU5hbmN5MRswGQYDVQQD
+                            ExJodHRwOi8vbmFuY3lmeC5vcmcwgZ8wDQYJKoZIhvcNAQEBBQADgY0AMIGJAoGB
+                            AMT4vOtIH9Fzad+8KCGjMPkkVpCtn+L5H97bnI3x+y3x5lY0WRsK8FyxVshY/7fv
+                            TDeeVKUWanmbMkQjgFRYffA3ep3/AIguswYdANiNVHrx0L7DXNDcgsjRwaa6JVgQ
+                            9iavlli0a80AF67FN1wfidlHCX53u3/fAjiSTwf7D+NBAgMBAAEwDQYJKoZIhvcN
+                            AQEFBQADgYEAh12A4NntHHdVMGaw+2umXkWqCOyAPuNhyBGUHK5vGON+VG0HPFaf
+                            8P8eMtdF4deBHkrfoWxRuGGn2tJzNpZLiEf23BAivEf36IqwfkVP7/zDwI+bjVXC
+                            k64Un2uN8ALR/wLwfJzHfOLPtsca7ySWhlv8oZo2nk0vR34asQiGJDQ=
+                            -----END CERTIFICATE-----
+                            ";
```
Comment (phillip-haydon) [inline src/Nancy.Testing/BrowserContext.cs]: Does this even work? Uploading a cert into AWS with whitespace at the start fails.

File: src/Nancy.Testing/project.json [hunk 2]
```diff
@@ -1,27 +1,30 @@
 ﻿{
-    "version": "2.0.0-*",
+  "version": "2.0.0-*",
```
Comment (phillip-haydon) [inline src/Nancy.Testing/project.json]: 4 space tabbing please. Fix this!

File: src/Nancy.Testing/BrowserContext.cs (hunk 1 above)
Comment (jchannon) [inline src/Nancy.Testing/BrowserContext.cs]: Yup this works. Although @thecodejunkie said maybe we should put this on
disk instead

**Comment by jchannon** [comment]
Once #2316 is merged I can rebase and the error list will be smaller

**Comment by phillip-haydon** [comment]
I can't restart the AppVeyor build.

**Comment by jchannon** [comment]
It's ok it won't compile anyway due to csquery not being compatible

**Comment by thecodejunkie** [comment]
Make sure to [WIP] this kind of pull-request

**Comment by jchannon** [comment]
Rebased.

Outstanding issues are CsQuery and Assembly.GetEntryAssembly

**Comment by jchannon** [comment]
I've reverted any certificate changes as its seems I'm too stupid to get it to work across csproj and dnx and for it pass tests on dnx. I end up with 

    System.Security.Cryptography.CryptographicException : Cannot find the requested object.

      Stack Trace:
           at System.Security.Cryptography.CryptographicException.ThrowCryptographicException(Int32 hr)
           at System.Security.Cryptography.X509Certificates.X509Utils._QueryCertBlobType(Byte[] rawData)
           at System.Security.Cryptography.X509Certificates.X509Certificate.LoadCertificateFromBlob(Byte[] rawData, Object password, X509KeyStorageFlags keyStorageFlags)
           at System.Security.Cryptography.X509Certificates.X509Certificate..ctor(Byte[] data)
        C:\Projects\Nancy\test\Nancy.Owin.Tests\AppBuilderExtensionsFixture.cs(52,0): at Nancy.Owin.Tests.AppBuilderExtensionsFixture.When_host_Nancy_via_IAppBuilder_should_read_X509Certificate2()

**Comment by jchannon** [comment]
Removed WIP title

**Comment by thecodejunkie** [comment]
I'll put this here so we have a reference: Right now `AngleSharp` is a PCL and as such we would either need to use `type: build` for the system assembly facades or stick with the version number but be aware of the caveat. Basically we can't put out a NuGet if we use a version number, because it would not play nice with `.csproj` see https://stackoverflow.com/questions/34136228/unable-to-resolve-assembly-reference-issue-without-frameworkassemblies/34139541#34139541 for more details

However, we know that `AngleSharp` is due to put out a version that targets the CoreCLR very soon

**Comment by thecodejunkie** [comment]
I'll change the `.GetTypeInfo().Assembly` calls to `.GetAssembly()` when I submit my PR for the functional tests

**Pull request merged**